torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -64,6 +65,32 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         prim::RequiresGradCheck /* (...)  -> (..., bool) */,
+         [](const Node* node) -> Operation {
+           std::vector<c10::optional<bool>> rg_props =
```

Comment:
Just to be sure, this `std::vector` is only constructed once, right, and then cached inside the lambda below?  I just want to make sure we're not allocating on every RequiresGradCheck.

Reply:
yup!

Reply:
I  don't think rg-props  should ever contain c10::nullopt, that wouldn't be checking anything at all. (it also makes it so we have to do an extra unwrap of the bool). Can we change `std::vector<c10::optional<bool>> ` ->  `std::vector<bool>` ?

Reply:
Yeah, we can, I just double check that Thomas indeed is conservatively setting `requiresGrad` to true if we it changed during profiling. Let me update it